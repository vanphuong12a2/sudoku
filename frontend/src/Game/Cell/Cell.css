.main-board input.finish {
    background: lightgreen;
}

.main-board input.writable {
    color: green;
}

.main-board input.readonly {
    cursor: pointer;
    color: black;
}

.main-board input.invalid {
    background: pink;
    border: 0;
    box-shadow: none;
}