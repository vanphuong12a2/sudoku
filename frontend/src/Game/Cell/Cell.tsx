import React, {ChangeEvent} from "react";

export const EMPTY_CELL = 0;

interface Props {
    cellData: number
    readOnly: boolean
    onCellChange: (newValue: number) => void
}

interface State {
    value: number
}

class Cell extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {value: props.cellData};
    }

    private handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        let newValue = event.target.value ? Number(event.target.value) : EMPTY_CELL;
        this.setState({value: newValue});
        this.props.onCellChange(newValue);
    };

    render() {
        return (
            <td>
                <input
                    className={this.props.readOnly ? 'readonly' : 'writable'}
                    maxLength={1}
                    readOnly={this.props.readOnly}
                    value={this.state.value}
                    onChange={this.handleChange}
                />
            </td>
        );
    }
}

export default Cell;