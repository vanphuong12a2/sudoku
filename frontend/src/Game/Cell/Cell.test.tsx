import React from 'react';
import {create} from 'react-test-renderer';
import Cell from './Cell';

describe('<Cell />', () => {

    describe('Snapshots', () => {

        it('renders Cell correctly', () => {

            const cell = create(
                <Cell
                    cellData={3}
                    readOnly={true}
                    onCellChange={jest.fn()}
                />);
            expect(cell.toJSON()).toMatchSnapshot();
        });

        it('renders empty Cell correctly', () => {

            const cell = create(
                <Cell
                    cellData={0}
                    readOnly={true}
                    onCellChange={jest.fn()}
                />);
            expect(cell.toJSON()).toMatchSnapshot();
        });
    });

});