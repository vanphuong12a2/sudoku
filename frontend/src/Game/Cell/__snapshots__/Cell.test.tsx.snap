// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Snapshots renders Cell correctly 1`] = `
<td>
  <input
    className="readonly invalid"
    maxLength={1}
    onChange={[Function]}
    readOnly={true}
    value={3}
  />
</td>
`;

exports[`<Cell /> Snapshots renders empty Cell correctly 1`] = `
<td>
  <input
    className="readonly"
    maxLength={1}
    onChange={[Function]}
    readOnly={true}
    value=""
  />
</td>
`;

exports[`<Cell /> Snapshots renders finish Cell correctly 1`] = `
<td>
  <input
    className="finish"
    maxLength={1}
    onChange={[Function]}
    readOnly={true}
    value=""
  />
</td>
`;
